### PR TITLE
Add deletion interfaces for VM, VMSS and interface clients

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind feature
/sig cloud-provider
/area provider/azure
/priority important-soon

**What this PR does / why we need it**:

Continue of https://github.com/kubernetes-sigs/cloud-provider-azure/issues/247: Add deletion interfaces for VM, VMSS and interface clients. Those interfaces would be used in cluster-autoscaler.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmssclient/azure_vmssclient.go [hunk 1]
```diff
@@ -359,3 +359,69 @@ func (page VirtualMachineScaleSetListResultPage) Values() []compute.VirtualMachi
...
+	response, rerr := c.armClient.PostResource(ctx, resourceID, "delete", vmInstanceIDs)
+	defer c.armClient.CloseResponse(ctx, response)
+	if rerr != nil {
+		klog.V(5).Infof("Received error in %s: resourceID: %s, error: %s", "vmss.deletevms.request", resourceID, rerr.Error())
+		return rerr
+	}
+
+	err := autorest.Respond(response, azure.WithErrorUnlessStatusCode(http.StatusOK, http.StatusAccepted))
+	if err != nil {
+		klog.V(5).Infof("Received error in %s: resourceID: %s, error: %s", "vmss.deletevms.respond", resourceID, rerr.Error())
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmssclient/azure_vmssclient.go]: shall we make it as V(2) ?

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmssclient/azure_vmssclient.go]: Those errors would also be logged at client consumers, so V(2) is not needed here.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**